low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -39,18 +39,30 @@ class ServerMap(ResourceMap):
     def __init__(self):
         """Initialize server"""
         super().__init__()
+        self.validate_server_type_support()
         self.sysfs = ToolFactory().get_instance('sysfs')
         self.sysfs.initialize()
         self.sysfs_base_path = self.sysfs.get_sysfs_base_path()
         self.cpu_path = self.sysfs_base_path + CPU_PATH
         self.server_frus = {
             'cpu': self.get_cpu_info,
             'platform_sensors': self.get_platform_sensors_info,
-            'memory': self.get_mem_info
+            'memory': self.get_mem_info,
+            'fans': self.get_fans_info
         }
         self._ipmi = IpmiFactory().get_implementor("ipmitool")
         self.platform_sensor_list = ['Temperature', 'Voltage', 'Current']
 
+    @staticmethod
+    def validate_server_type_support():
+        """Check for supported server type."""
+        server_type = Conf.get(GLOBAL_CONF, NODE_TYPE_KEY).lower()
+        supported_types = ["virtual", "physical"]
```

Comment:
Only server type "physical" is supported as per existing generate_health_view script.
@satish-seagate, please confirm.

Reply:
Yes, @mariyappanp. we have decided in the very long back in one of the meetings that we can support only physical servers and at that time @indrajitzagade  and @cdeshmukh were present.
Need to confirm with @cdeshmukh that we should follow the same way to support only physical server or we have to enhance our scope for virtual server too.
@cdeshmukh Please confirm.
FYI - @ujjwalpl @thavanathan

Reply:
Yes, physical server is the main support we need. For VM, physical resources health should come NULL, like server FAN here.
On VM, mainly server disks(virtual), cpu (virtual), memory(virtual) and sw services health we can give i believe, which we can add up in future.
Currently limit to physical server.

@ujjwalpl please state if any concerns

Reply:
Checked with my patch #427 (storage disks), @cdeshmukh 
I generated compute and storage health information. Looks good with virtual type. too, request processed successfully. Health info of unavailable FRUs return `[ ]` as response. 

At this time on compute, cpu, memory, platform_sensors are validated. cpu, memory gives response with values and platform_sensors returns 'null'. (this will be corrected to [ ])

But what is the node structure for virtual type server health? As per current health schema, I see only hw, sw under compute. @cdeshmukh, @satish-seagate

Reply:
hmm going point, virtual parent needs to be incorporated seperately, to add virtual cpu, memory, disks under it and hw can be blank. sw will have regular data, what we get on physical server too

Reply:
Kept 'physical' server type for now and added a TODO to add support for virtual server type as well

Reply:
Ok